lib/cppcheck.cpp

```diff
@@ -1107,9 +1105,7 @@ void CppCheck::checkNormalTokens(const Tokenizer &tokenizer)
     if (mSettings.checks.isEnabled(Checks::unusedFunction) && !mSettings.buildDir.empty()) {
         unusedFunctionsChecker.parseTokens(tokenizer, tokenizer.list.getFiles().front().c_str(), mSettings);
     }
-    if (mSettings.checks.isEnabled(Checks::unusedFunction) &&
-        mSettings.useSingleJob() &&
-        mSettings.buildDir.empty()) {
+    if (mSettings.isUnusedFunctionCheckEnabled() && mSettings.buildDir.empty()) {
```

Comment:
hmm.. I don't understand why we required that the buildDir is empty. Don't we want to run this checking here for some reason if the build dir is empty..

Reply:
I have no idea. Currently I am keeping the current behavior intact. I will look into this when I look at the build directory related false positives. But that requires adding a lot more general testing of the build directory first.